dashboard/src/shared/types.ts

```diff
@@ -398,37 +398,24 @@ export interface IKubeState {
   kindsError?: Error;
 }
 
-export interface IBasicFormParam {
-  path: string;
-  type?: "string" | "number" | "integer" | "boolean" | "object" | "array" | "null" | "any";
-  value?: any;
-  title?: string;
-  minimum?: number;
-  maximum?: number;
-  render?: string;
-  description?: string;
-  customComponent?: object;
+// We extend the JSONSchema properties to include the default/deployed values as well as
+// other useful information for rendering each param in the UI
+export type IBasicFormParam = JSONSchemaType<any> & {
+  key: string;
+  title: string;
+  hasProperties: boolean;
+  params?: IBasicFormParam[];
   enum?: string[];
-  hidden?:
-    | {
-        event: DeploymentEvent;
-        path: string;
-        value: string;
-        conditions: Array<{
-          event: DeploymentEvent;
-          path: string;
-          value: string;
-        }>;
-        operator: string;
-      }
-    | string;
-  children?: IBasicFormParam[];
-}
-export interface IBasicFormSliderParam extends IBasicFormParam {
-  sliderMin?: number;
-  sliderMax?: number;
-  sliderStep?: number;
-  sliderUnit?: string;
+  defaultValue: any;
+  deployedValue: any;
+  currentValue: any;
+  schema: JSONSchemaType<any>;
+  isCustomComponent?: boolean;
+};
+
+export interface IAjvValidateResult {
```

Comment:
Why using this "criptic"" name? :D

Reply:
No strong opinion on the naming, but since it is the result of performing the `validate` function on an`ajv` (the library for handling schemas) object, this name is the best I could think of. Suggestions? :P

Reply:
Well, `ErrorObject`, one of the used types, is indeed imported from the Ajv library, so we would be kinda coupled to `ajv` anyway? But yep, at least we should add a comment explaining what is this interface for.

Reply:
You are right. Fine leaving it as is with a comment.